crates/ledger-types/src/state_accessor.rs

```diff
@@ -0,0 +1,50 @@
+use strata_acct_types::{AccountId, AccountSerial, AcctResult};
+
+use crate::{
+    account::{AccountTypeState, IAccountState},
+    global_state::IGlobalState,
+};
+
+/// Opaque interface for manipulating the chainstate, for all of the parts
+/// directly under the toplevel state.
+///
+/// This exists because we want to make this generic across the various
+/// different contexts we'll be manipulating state.
+pub trait StateAccessor {
+    /// Type representing the global chainstate.
+    type GlobalState: IGlobalState;
```

Comment:
This is the toplevel state right?

Also, Is it really necessary to have trait bounds for `GlobalState` and `AccountState`?Because neither's internals are being used as argument or return types in the trait methods.

Basically, I am not sure if trait bounds for Global State and Account State are needed because it seems to me that accessing/modifying state is being done via this accessor itself, it should be sufficient to define those getters/setters/modifiers in the `StateAccessor` trait. I would even say that with mutable getters for both states, we might even not need to have fine grained getters/setters. I might be wrong or missing something though.